app/controllers/admin/users_controller.rb

```diff
@@ -44,22 +44,16 @@ def destroy
       role = params[:role].to_sym
       resource_type = params[:resource_type]
 
-      if role == :super_admin
-        flash[:danger] = "Super Admin roles cannot be removed."
-        redirect_to "/admin/users/#{params[:id]}/edit" and return
-      end
-
       @user = User.find(params[:user_id])
 
-      if @user.id == current_user.id
-        flash[:danger] = "Admins cannot remove roles from themselves."
-      elsif role == :single_resource_admin && !resource_type.safe_constantize.nil?
-        User.find(params[:user_id]).remove_role(role, resource_type.safe_constantize)
-        flash[:success] = "Role: #{role.to_s.humanize.titlecase} has been successfully removed from the user!"
-      elsif User.find(params[:user_id]).remove_role(role)
+      response = ::Users::RemoveRole.call(user: @user,
+                                          role: role,
+                                          resource_type: resource_type,
+                                          current_user: current_user)
```

Comment:
Nitpick: having named arguments named `user` and `current_user` could be slightly confusing. Should we name them `user` and `admin` respectively?

Reply:
Hm, you bring up a great point! I thought about this while writing the spec for this service and think it could be potentially confusing either way. That said, I don't feel strongly one way or another and think that since you (with fresh eyes!) think it could be slightly confusing, then others might find it slightly confusing, so I've replaced `current_user` with `admin` in all necessary places. 😊